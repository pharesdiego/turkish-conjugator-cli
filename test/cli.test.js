const { exec } = require("child_process");
const path = require("path");
const file = path.resolve(process.cwd(), "index.js");

test("CLI gitmek", (done) => {
  exec(file + " gitmek", (error, stdout) => {
    expect(stdout).toMatchSnapshot();
    done();
  });
});

test("CLI gitmek --aorist", (done) => {
  exec(file + " gitmek --aorist", (error, stdout) => {
    expect(stdout).toMatchSnapshot();
    done();
  });
});

test("CLI gitmek --gerund", (done) => {
  exec(file + " gitmek --gerund", (error, stdout) => {
    expect(stdout).toMatchSnapshot();
    done();
  });
});

test("CLI no valid verb", (done) => {
  exec(file + " gitmasdqwd", (error, stdout) => {
    expect(stdout).toMatchSnapshot();
    done();
  });
});

test("CLI ignore no valid options", (done) => {
  exec(file + " gitmek --hola --yes --aorist", (error, stdout) => {
    expect(stdout).toMatchSnapshot();
    done();
  });
});