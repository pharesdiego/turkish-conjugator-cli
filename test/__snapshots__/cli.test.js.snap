// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI gitmek --aorist 1`] = `
"╔═══════════════════════════╤═══════════════════════════╗
║ [31mG[39m[33mI[39m[32mT[39m[34mM[39m[35mE[39m[31mK[39m                    │                           ║
╟───────────────────────────┼───────────────────────────╢
║ [1mAorist[22m                    │                           ║
╟───────────────────────────┼───────────────────────────╢
║ [32mPositive[39m                  │ [35mNegative[39m                  ║
╟───────────────────────────┼───────────────────────────╢
║ giderim                   │ gitmem                    ║
║ gidersin                  │ gitmezsin                 ║
║ gider                     │ gitmez                    ║
║ gideriz                   │ gitmeyiz                  ║
║ gidersiniz                │ gitmezsiniz               ║
║ giderler                  │ gitmezler                 ║
╚═══════════════════════════╧═══════════════════════════╝

"
`;

exports[`CLI gitmek --gerund 1`] = `
"╔═══════════════════════════╤═══════════════════════════╗
║ [31mG[39m[33mI[39m[32mT[39m[34mM[39m[35mE[39m[31mK[39m                    │                           ║
╟───────────────────────────┼───────────────────────────╢
║ [1mGerund[22m                    │                           ║
╟───────────────────────────┼───────────────────────────╢
║ [32mPositive[39m                  │ [35mNegative[39m                  ║
╟───────────────────────────┼───────────────────────────╢
║ gidiyorum                 │ gitmiyorum                ║
║ gidiyorsun                │ gitmiyorsun               ║
║ gidiyor                   │ gitmiyor                  ║
║ gidiyoruz                 │ gitmiyoruz                ║
║ gidiyorsunuz              │ gitmiyorsunuz             ║
║ gidiyorlar                │ gitmiyorlar               ║
╚═══════════════════════════╧═══════════════════════════╝

"
`;

exports[`CLI gitmek 1`] = `
"╔═══════════════════════════╤═══════════════════════════╗
║ [31mG[39m[33mI[39m[32mT[39m[34mM[39m[35mE[39m[31mK[39m                    │                           ║
╟───────────────────────────┼───────────────────────────╢
║ [1mAorist[22m                    │                           ║
╟───────────────────────────┼───────────────────────────╢
║ [32mPositive[39m                  │ [35mNegative[39m                  ║
╟───────────────────────────┼───────────────────────────╢
║ giderim                   │ gitmem                    ║
║ gidersin                  │ gitmezsin                 ║
║ gider                     │ gitmez                    ║
║ gideriz                   │ gitmeyiz                  ║
║ gidersiniz                │ gitmezsiniz               ║
║ giderler                  │ gitmezler                 ║
╟───────────────────────────┼───────────────────────────╢
║ [1mGerund[22m                    │                           ║
╟───────────────────────────┼───────────────────────────╢
║ [32mPositive[39m                  │ [35mNegative[39m                  ║
╟───────────────────────────┼───────────────────────────╢
║ gidiyorum                 │ gitmiyorum                ║
║ gidiyorsun                │ gitmiyorsun               ║
║ gidiyor                   │ gitmiyor                  ║
║ gidiyoruz                 │ gitmiyoruz                ║
║ gidiyorsunuz              │ gitmiyorsunuz             ║
║ gidiyorlar                │ gitmiyorlar               ║
╟───────────────────────────┼───────────────────────────╢
║ [1mFuture[22m                    │                           ║
╟───────────────────────────┼───────────────────────────╢
║ [32mPositive[39m                  │ [35mNegative[39m                  ║
╟───────────────────────────┼───────────────────────────╢
║ gideceğim                 │ gitmeyeceğim              ║
║ gideceksin                │ gitmeyeceksin             ║
║ gidecek                   │ gitmeyecek                ║
║ gideceğiz                 │ gitmeyeceğiz              ║
║ gideceksiniz              │ gitmeyeceksiniz           ║
║ gidecekler                │ gitmeyecekler             ║
╟───────────────────────────┼───────────────────────────╢
║ [1mSimple Past[22m               │                           ║
╟───────────────────────────┼───────────────────────────╢
║ [32mPositive[39m                  │ [35mNegative[39m                  ║
╟───────────────────────────┼───────────────────────────╢
║ gittim                    │ gitmedim                  ║
║ gittin                    │ gitmedin                  ║
║ gitti                     │ gitmedi                   ║
║ gittik                    │ gitmedik                  ║
║ gittiniz                  │ gitmediniz                ║
║ gittiler                  │ gitmediler                ║
╟───────────────────────────┼───────────────────────────╢
║ [1mMiş Past[22m                  │                           ║
╟───────────────────────────┼───────────────────────────╢
║ [32mPositive[39m                  │ [35mNegative[39m                  ║
╟───────────────────────────┼───────────────────────────╢
║ gitmişim                  │ gitmemişim                ║
║ gitmişsin                 │ gitmemişsin               ║
║ gitmiş                    │ gitmemiş                  ║
║ gitmişiz                  │ gitmemişiz                ║
║ gitmişsiniz               │ gitmemişsiniz             ║
║ gitmişler                 │ gitmemişler               ║
╟───────────────────────────┼───────────────────────────╢
║ [1mPast Continuous[22m           │                           ║
╟───────────────────────────┼───────────────────────────╢
║ [32mPositive[39m                  │ [35mNegative[39m                  ║
╟───────────────────────────┼───────────────────────────╢
║ gidiyordum                │ gitmiyordum               ║
║ gidiyordun                │ gitmiyordun               ║
║ gidiyordu                 │ gitmiyordu                ║
║ gidiyorduk                │ gitmiyorduk               ║
║ gidiyordunuz              │ gitmiyordunuz             ║
║ gidiyordular              │ gitmiyordular             ║
╟───────────────────────────┼───────────────────────────╢
║ [1mPresent Indefinite[22m        │                           ║
╟───────────────────────────┼───────────────────────────╢
║ [32mPositive[39m                  │ [35mNegative[39m                  ║
╟───────────────────────────┼───────────────────────────╢
║ gidermişim                │ gitmezmişim               ║
║ gidermişsin               │ gitmezmişsin              ║
║ gidermiş                  │ gitmezmiş                 ║
║ gidermişiz                │ gitmezmişiz               ║
║ gidermişsiniz             │ gitmezmişsiniz            ║
║ gidermişler               │ gitmezmişler              ║
╟───────────────────────────┼───────────────────────────╢
║ [1mPresent Continuous Indefi[22m │                           ║
║ [1mnite[22m                      │                           ║
╟───────────────────────────┼───────────────────────────╢
║ [32mPositive[39m                  │ [35mNegative[39m                  ║
╟───────────────────────────┼───────────────────────────╢
║ gidiyormuşum              │ gitmiyormuşum             ║
║ gidiyormuşsun             │ gitmiyormuşsun            ║
║ gidiyormuş                │ gitmiyormuş               ║
║ gidiyormuşuz              │ gitmiyormuşuz             ║
║ gidiyormuşsunuz           │ gitmiyormuşsunuz          ║
║ gidiyormuşlar             │ gitmiyormuşlar            ║
╟───────────────────────────┼───────────────────────────╢
║ [1mFuture Indefinite[22m         │                           ║
╟───────────────────────────┼───────────────────────────╢
║ [32mPositive[39m                  │ [35mNegative[39m                  ║
╟───────────────────────────┼───────────────────────────╢
║ gidecekmişim              │ gitmeyecekmişim           ║
║ gidecekmişsin             │ gitmeyecekmişsin          ║
║ gidecekmiş                │ gitmeyecekmiş             ║
║ gidecekmişiz              │ gitmeyecekmişiz           ║
║ gidecekmişsiniz           │ gitmeyecekmişsiniz        ║
║ gidecekmişler             │ gitmeyecekmişler          ║
╟───────────────────────────┼───────────────────────────╢
║ [1mPotential[22m                 │                           ║
╟───────────────────────────┼───────────────────────────╢
║ [32mPositive[39m                  │ [35mNegative[39m                  ║
╟───────────────────────────┼───────────────────────────╢
║ gidebilirim               │ gidemem                   ║
║ gidebilirsin              │ gidemezsin                ║
║ gidebilir                 │ gidemez                   ║
║ gidebiliriz               │ gidemeyiz                 ║
║ gidebilirsiniz            │ gidemezsiniz              ║
║ gidebilirler              │ gidemezler                ║
╚═══════════════════════════╧═══════════════════════════╝

"
`;

exports[`CLI ignore no valid options 1`] = `
"╔═══════════════════════════╤═══════════════════════════╗
║ [31mG[39m[33mI[39m[32mT[39m[34mM[39m[35mE[39m[31mK[39m                    │                           ║
╟───────────────────────────┼───────────────────────────╢
║ [1mAorist[22m                    │                           ║
╟───────────────────────────┼───────────────────────────╢
║ [32mPositive[39m                  │ [35mNegative[39m                  ║
╟───────────────────────────┼───────────────────────────╢
║ giderim                   │ gitmem                    ║
║ gidersin                  │ gitmezsin                 ║
║ gider                     │ gitmez                    ║
║ gideriz                   │ gitmeyiz                  ║
║ gidersiniz                │ gitmezsiniz               ║
║ giderler                  │ gitmezler                 ║
╚═══════════════════════════╧═══════════════════════════╝

"
`;

exports[`CLI no valid verb 1`] = `
"Please write a turkish verb
"
`;
